pandas/core/arrays/categorical.py

```diff
@@ -2506,16 +2505,90 @@ class CategoricalAccessor(PandasDelegate, PandasObject, NoNewAttributesMixin):
 
     Examples
     --------
+    >>> s = pd.Series(list("aabc")).astype("category")
+    >>> s
+    0    a
+    1    a
+    2    b
+    3    c
+    dtype: category
+    Categories (3, object): [a, b, c]
+
     >>> s.cat.categories
-    >>> s.cat.categories = list('abc')
-    >>> s.cat.rename_categories(list('cab'))
-    >>> s.cat.reorder_categories(list('cab'))
-    >>> s.cat.add_categories(['d','e'])
+    Index(['a', 'b', 'c'], dtype='object')
+
+    >>> s.cat.categories = list("bcd")
+    >>> s
+    0    b
+    1    b
+    2    c
+    3    d
+    dtype: category
+    Categories (3, object): [b, c, d]
+
+    >>> s.cat.rename_categories(list("abc"))
+    0    a
+    1    a
+    2    b
+    3    c
+    dtype: category
+    Categories (3, object): [a, b, c]
+
+    >>> s.cat.reorder_categories(list("cdb"))
+    0    b
+    1    b
+    2    c
+    3    d
+    dtype: category
+    Categories (3, object): [c, d, b]
+
+    >>> s.cat.add_categories(["e", "f"])
+    0    b
+    1    b
+    2    c
+    3    d
+    dtype: category
+    Categories (5, object): [b, c, d, e, f]
+
     >>> s.cat.remove_categories(['d'])
+    0      b
+    1      b
+    2      c
+    3    NaN
+    dtype: category
+    Categories (2, object): [b, c]
+
     >>> s.cat.remove_unused_categories()
```

Comment:
Can you assign the above one with `add_categories` to eg `s2` and use that here? (so the `remove_unused_categories` actually has an effect)